synapse/storage/database.py

```diff
@@ -1226,9 +1253,9 @@ async def simple_upsert_many(
         self,
         table: str,
         key_names: Collection[str],
-        key_values: Collection[Iterable[Any]],
+        key_values: Collection[Collection[Any]],
         value_names: Collection[str],
-        value_values: Iterable[Iterable[Any]],
+        value_values: Collection[Collection[Any]],
```

Comment:
Does `simple_upsert_many_txn` need to match these?

Reply:
I don't think so, because `simple_upsert_many_txn` only goes through the iterables once.